clap_complete/src/dynamic/completer.rs

```diff
@@ -278,9 +282,9 @@ fn complete_subcommand(value: &str, cmd: &clap::Command) -> Vec<OsString> {
     let mut scs = crate::generator::utils::subcommands(cmd)
         .into_iter()
         .filter(|x| x.0.starts_with(value))
-        .map(|x| OsString::from(&x.0))
+        .map(|x| (OsString::from(&x.0), x.2))
         .collect::<Vec<_>>();
     scs.sort();
-    scs.dedup();
+    // TODO: is this needed, doesn't work with styled string scs.dedup();
```

Comment:
Why doesn't this work?  Looks like it just needs `PartialEq` which `StyledStr` impls.

I don't have the time to analyze this for now, so I'd prefer we fix this

Please call these out yourself.  if I had missed this, I would have merged this as-is